queries/prolog/highlights.scm

```diff
@@ -0,0 +1,59 @@
+(comment) @comment
+
+(atom) @constant
+
+((atom) @boolean
+  (#eq? @boolean "true"))
+
+((atom) @boolean
+  (#eq? @boolean "false"))
+
+(functional_notation
+  function: (atom) @function.call)
+
+[
+  (integer)
+  (negative_number
+    (integer))
+] @number
```

Comment:
```suggestion
(integer) @number
```

Reply:
Oh I see.

Reply:
@lucario387 are you talking about making `-` part of `integer` and removing the `negative_number` non-terminal altogether? Although I do agree that it is very weird to use `integer` to only refer to non-negative integer numbers, the current state of the parser is meant to reflect how the ISO specification defines the language. The currenty query is built keeping in mind we do not want to change the parser.

Reply:
Yes, that's what I meant. It's kindof weird for the ISO spec of a language to specifically denote negative number though. Is there any special usage for it?

Reply:
I cannot find an explicit motivation anywhere, sorry. Could this has been done for unification/pattern matching, maybe? This way, `delta_is(-1)` can match `delta_is(-x)` for example. In the original specs, negative numbers are specified as:

```
6.3.1.2 Negative numbers
term = name, integer;
term = name, float number;
(Condition: name is "-")
```

So the non-terminal "name" is used, but then a side condition is specified.

I personally think sticking to the language itself is beneficial in this case, but I am still open to rewriting this part of the parser as long as I can document the change with a valid reason.

Reply:
In general, sticking too closely to the language spec is not a good idea; often a more pragmatic approach that focuses on the end goal (which for us is syntax highlighting) leads to a more robust and performant parser. [(Github says so.)](https://github.com/github/semantic/blob/3baf0b47ff2117a4724771687d458151ce2b92a7/docs/grammar-development-guide.md#rely-on-language-specs-but-not-too-much)

That being said, in the end it's your decision; this is not a critical blocker.

Reply:
I understand! You make a valid point. I'll need some time to make the changes, and then ping you all when I am ready. Thanks.